[ruishanteo] iP

ruishanteo opened the pull request:
# Bot

> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Bot frees your mind of having to remember things you need to do. It's:

- **Text-based**
- **Easy to learn**
- **~~FAST~~** _SUPER FAST_ to use

All you need to do is:

1. Download it from [here](https://github.com/ruishanteo/ip)
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
5. And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ruishanteo committed:
completed level 0

ruishanteo committed:
completed level 1

ruishanteo committed:
completed level 2

ruishanteo committed:
completed level 3

ruishanteo committed:
completed level 4

ruishanteo committed:
completed A-TextUiTesting

ruishanteo committed:
completed level 5

ruishanteo committed:
completed level 6

ruishanteo committed:
Add save format line

ruishanteo committed:
Add save tasks

ruishanteo committed:
Update .gitignore

ruishanteo committed:
Complete Level 7

ruishanteo committed:
Update test cases

ruishanteo committed:
Add DateTime object

ruishanteo committed:
Merge branch 'branch-Level-8'

ruishanteo committed:
Update test case

ruishanteo committed:
Implement command classes

ruishanteo committed:
Update Duke.java

ruishanteo committed:
Add Parser

ruishanteo committed:
Add Storage

ruishanteo committed:
Add TaskList

ruishanteo committed:
Add Ui

ruishanteo committed:
Merge branch 'branch-A-MoreOOP'

ruishanteo committed:
Add packages and refactored

ruishanteo committed:
Update files

ruishanteo committed:
Merge branch 'branch-A-Packages'

ruishanteo committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

ruishanteo committed:
Update build file

ruishanteo committed:
Merge branch 'branch-A-Gradle'

ruishanteo committed:
Add test files

ruishanteo committed:
Merge branch 'branch-A-JUnit'

ruishanteo committed:
Add javadoc

ruishanteo committed:
Merge branch 'branch-A-Javadoc'

ruishanteo committed:
Fix style

ruishanteo committed:
Merge branch 'branch-A-CodingStandard'

ruishanteo committed:
Add find functionalities

ruishanteo committed:
Merge branch-Level-9

ruishanteo committed:
Add import

ruishanteo committed:
Add checkstyle

ruishanteo committed:
Merge branch 'branch-A-CheckStyle'

ruishanteo committed:
Update build file

ruishanteo committed:
Add javafx application

ruishanteo committed:
Add controllers and fxml files

ruishanteo committed:
Add images for users

ruishanteo committed:
Merge branch 'branch-Level-10'

Choonyan02 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,147 @@
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+
+public class ParserTest {
```
I like the untrivial test cases given here!

Choonyan02 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,129 @@
+        currentMessage += "There are no results matching your keyword.";
+    }
+
+    public String getCurrentMessage() {
+        String temp = currentMessage;
+        currentMessage = "";
+        return temp;
+    }
```
I like this method, I had to change all my method to return String instead of void 😄

Choonyan02 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,216 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    private Task readLine(String line) {
+        String[] split = line.split(" \\| ");
+        String type = split[0];
+        boolean isDone = split[1].equals("1");
+        String description = split[2];
+        Task task = null;
+        if (type.equals("T")) {
+            task = this.addToDo(description);
+        } else if (type.equals("E")) {
+            task = this.addEvent(description, split[3], split[4]);
+        } else if (type.equals("D")) {
+            task = this.addDeadline(description, split[3]);
+        }
+
+        if (isDone) {
+            task.markAsDone();
+        } else {
+            task.markAsUndone();
+        }
+        return task;
+    }
```
Perhaps you can add a JavaDoc for this method?

Choonyan02 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,129 @@
+    /**
+     * Displays a welcome message along with the Duke logo.
+     */
+    public void showWelcome() {
+        String logo =
+                ".______     ______   .___________.\n"
+                        + "|   _  \\   /  __  \\  |           |\n"
+                        + "|  |_)  | |  |  |  | `---|  |----`\n"
+                        + "|   _  <  |  |  |  |     |  |     \n"
+                        + "|  |_)  | |  `--'  |     |  |     \n"
+                        + "|______/   \\______/      |__|     \n";
+
+        currentMessage += "Hello I'm\n" + logo + "What can I do for you?";
+    }
```
I like your unique welcome message haha

Choonyan02 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,158 @@
+                throw new DukeException("The keyword to find cannot be empty.");
+            }
+            return new FindCommand(subInput);
+        } else if (input.contains("todo")) {
```
Perhaps you can do input.startsWith("todo")? Same for the others too. Both works though!

Choonyan02 reviewed (commented):
Overall, LGTM! Code is clean and good OOP

wujy28 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,216 @@
+     */
+    public Deadline addDeadline(String description, String by) {
+        Deadline deadline = new Deadline(description, LocalDateTime.parse(by,
+                                                                          DateTimeFormatter.ofPattern(
```
I'm not sure whether this type of line break is allowed? Might be safer to just break once with an indentation of 8 spaces from the front of the line 😊

wujy28 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,49 @@
+    @Override
+    public String toSaveLine() {
+        return String.format("E | %d | %s | %s | %s", isDone ? 1 : 0, description,
+                             start.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
```
Perhaps the line break should be 8 spaces indented from the front of the line too? Same goes for the next line :)

wujy28 commented on src/main/resources/images/DaDuke.png:
Cute picture XD

wujy28 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,147 @@
+public class ParserTest {
+
+    @Test
+    public void testParseByeCommand() {
```
I think test method names should be formatted as `featureUnderTest_testScenario_expectedBehavior()`?

wujy28 reviewed (commented):
Good job on all the tasks and also for going the extra mile to create the classes for `Command`!
Very minimal issues, mainly just the naming for test methods
Can't wait to talk to bot 🤣

ruishanteo committed:
Add assertions at initialization

Elements are assumed to not be null.

Add assertions to assert that they are not null before use.

ruishanteo committed:
Improve code quality

There are some coding standard improvments to be made as per PR comments.

Code readibility can be improved according to good practices.

Change structure of presentation of code.

ruishanteo committed:
Improve code quality

There are some coding standard improvments to be made as per PR comments.

Code readibility can be improved according to good practices.

Change structure of presentation of code.

ruishanteo committed:
Merge pull request #1 from ruishanteo/branch-A-Assertions

Add assertions at initialization

ruishanteo committed:
Merge branch 'branch-A-CodeQuality' of github.com:ruishanteo/ip into branch-A-CodeQuality

ruishanteo committed:
Merge pull request #2 from ruishanteo/branch-A-CodeQuality

Branch A-CodeQuality

ruishanteo committed:
Set up CI

ruishanteo committed:
Merge branch 'master' of github.com:ruishanteo/ip

ruishanteo committed:
Fix checkstyle error

ruishanteo committed:
Update packages

Many classes of the same kind are created.

Classes with similar functionalities are categorised together.

Let's,
* Create new packages to organise the classes
* Refactor the code to accommodate the changes

ruishanteo committed:
Add edit task functionalities

Tasks cannot be edited without being deleted.

Users need to delete and create a new task to update list.

Let's
* Create new EditCommand to execute editing functionalities.
* Relocate logic of parsing tasks so that new task can be created.
* Update TaskList to include editTask method.

ruishanteo committed:
Fix checkstyle errors

ruishanteo committed:
Merge branch 'branch-C-Update'

ruishanteo committed:
Update formatting

ruishanteo committed:
Update task details editing

ruishanteo committed:
Merge branch 'branch-C-Update-1'

kristayeo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,236 @@
+        } else {
+            ui.findNoMessage();
+        }
+    }
```
LGTM! Naming of the classes is clear and code is neat

ruishanteo committed:
Improve code quality

ruishanteo committed:
Add user guide

ruishanteo committed:
Add chatbot name

ruishanteo committed:
Add new line character

ruishanteo committed:
Add example usage of edit

ruishanteo committed:
Update readme
